packages/network-controller/src/rpc-service/rpc-service.ts

```diff
@@ -485,26 +488,55 @@ export class RpcService implements AbstractRpcService {
     } catch (error) {
       if (error instanceof HttpError) {
         const status = error.httpStatus;
-        if (status === 405) {
-          throw rpcErrors.methodNotFound();
+        if (status === 401) {
+          throw new JsonRpcError(-33100, 'Unauthorized.', {
+            httpStatus: status,
+          });
+        }
+        if (status === 402 || status === 404) {
+          throw rpcErrors.resourceUnavailable({
+            message: `RPC endpoint server error (HTTP ${status})`,
+            data: {
+              httpStatus: status,
+            },
+          });
+        }
+        if (status === 405 || status === 501) {
```

Comment:
Interpreting 405 as a JSON-RPC "method not found" error doesn't seem right to me. 405 means that [the server did not know what to do with the HTTP method sent](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/405) (so the JSON-RPC request was not even processed). 

Maybe we ought to interpret this as `invalidRequest`?